Armor pivots & misc fixes

omoflop opened the pull request:
+ Added pivot points for armor
* Fixed the vanilla armor model api bug from my last pr

+ Added a textbox to display the backend's motd

+ Formatted comments to consistently start with a space ( ex: "// text", previously: "//text" )

- Removed isCute and getIP from the api

+ Fixed typo in ban message

omoflop committed:
Refactors + Armor pivots

+ Armor placement is now be controlled with pivot points
+ Un-enum'd the Wizard (Support for addons will need more work)
+ Refactored VanillaModelAPI to make more sense
+ Tweaked some syntax in VanillaModelProvider
+ Added a wizard

omoflop committed:
Merge branch 'FiguraMC:1.20' into 1.20

omoflop committed:
lol

+ Formatted comments to have a space before any other text
+ Removed getIP() and isCute() functions from api
+ Readded vanilla armor api (oops)

omoflop committed:
motd thing

+ Added a spot for the backend motd
+ Fixed some broken urls (oops)

omoflop committed:
Merge branch 'FiguraMC:1.20' into 1.20

omoflop committed:
i accidentally left one line of text in here oml

omoflop committed:
Merge branch '1.20' of https://github.com/omoflop/Figura into 1.20

Covkie commented:
no way floppa you get 5 dollars from skype dot com

lenrik1589 commented on common/src/main/java/org/figuramc/figura/backend2/NetworkStuff.java:
```diff
@@ -194,6 +203,13 @@ public static void connect(String token) {
+    private static void fetchMOTD() {
+        queueString(Util.NIL_UUID, HttpAPI::getMotd, (code, data) -> {
+            responseDebug("motd", code, data);
+            motd = Component.Serializer.fromJson(data);
```
TextUtils.tryParseJson()

lenrik1589 commented on common/src/main/java/org/figuramc/figura/FiguraMod.java:
```diff
@@ -38,7 +38,7 @@ public class FiguraMod {
     public static final FiguraModMetadata METADATA = FiguraModMetadata.getMetadataForMod(MOD_ID);
     public static final Version VERSION = new Version(PlatformUtils.getModVersionString());
-    public static final boolean DEBUG_MODE = Math.random() + 1 < 0;
+    public static final boolean DEBUG_MODE = false;
```
This was done to prevent compiler optimizations

lenrik1589 commented on common/src/main/java/org/figuramc/figura/FiguraMod.java:
```diff
@@ -85,27 +85,27 @@ public static void tick() {
     }
 
-    // -- Helper Functions -- //
+    // -- Helper Functions -- // 
```
why is there a trailing space on each comment of this style :skull:

lenrik1589 commented on common/src/main/java/org/figuramc/figura/gui/FiguraGui.java:
```diff
@@ -44,11 +44,11 @@ public static void onRender(GuiGraphics guiGraphics, float tickDelta, CallbackIn
         FiguraMod.pushProfiler(FiguraMod.MOD_ID);
 
-        //render aperdoll
+        // render paperdoll
```
spelling? never heard of!
\- `f`
(thank you :heart:)

lenrik1589 commented on common/src/main/java/org/figuramc/figura/backend2/NetworkStuff.java:
```diff
@@ -7,6 +7,9 @@
 import net.minecraft.nbt.NbtIo;
+import net.minecraft.network.chat.Component;
+import net.minecraft.network.chat.FormattedText;
+import net.minecraft.server.commands.TellRawCommand;
```
what are all the imports for?

lenrik1589 commented on common/src/main/java/org/figuramc/figura/backend2/websocket/WebsocketThingy.java:
```diff
@@ -34,7 +34,7 @@ public class WebsocketThingy extends WebSocketClient {
         put(3000, "Unauthorized");
         put(4000, "Re-Auth");
-        put(4001, "Bananed ^.^");
+        put(4001, "Banned");
```
(this would count as an "easteregg")

lenrik1589 commented on common/src/main/java/org/figuramc/figura/gui/screens/ProfileScreen.java:
```diff
@@ -32,7 +32,7 @@ public void init() {
     @Override
     public void renderOverlays(GuiGraphics gui, int mouseX, int mouseY, float delta) {
-        //UIHelper.highlight(stack, button, TextUtils.tryParseJson("{\"text\":\"🦐🦐🦐🦐\",\"font\":\"figura:emojis\"}"));
+        // UIHelper.highlight(stack, button, TextUtils.tryParseJson("{\"text\":\"🦐🦐🦐🦐\",\"font\":\"figura:emojis\"}"));
```
:woozy_face: (was this literally just "replace `(?<=//)([^ ])` with ` $1`"?)

lenrik1589 commented on common/src/main/java/org/figuramc/figura/gui/screens/WardrobeScreen.java:
```diff
@@ -2,11 +2,10 @@
 import net.minecraft.ChatFormatting;
 import net.minecraft.client.Minecraft;
+import net.minecraft.client.gui.components.FittingMultiLineTextWidget;
+import net.minecraft.client.gui.screens.ChatScreen;
```
more stray imports and * import?

lenrik1589 commented on common/src/main/java/org/figuramc/figura/gui/widgets/BackendMessageWidget.java:
a little confused by this class?

lenrik1589 commented on common/src/main/java/org/figuramc/figura/gui/widgets/lists/AvatarWizardList.java:
```diff
@@ -15,12 +15,15 @@
 import java.util.List;
 import java.util.Map;
 
+import static org.figuramc.figura.wizards.WizardEntry.Type.CATEGORY;
```
stray, game of the year /lh

lenrik1589 commented on common/src/main/java/org/figuramc/figura/lua/FiguraLuaRuntime.java:
```diff
@@ -39,8 +39,8 @@
-    //Global API instances
-    //---------------------------------
+    // Global API instances
+    // ---------------------------------
```
this is a delimer, shouldn't have spaces there

lenrik1589 commented on common/src/main/java/org/figuramc/figura/lua/api/entity/EntityAPI.java:
```diff
@@ -140,12 +140,6 @@ public String getType() {
-    public boolean isCute() {
-        return true;
-    }
-
```
:pensive: so it seems, not every creature can be cute (~~tho, kinda makes sense with `f`~~)

lenrik1589 commented on common/src/main/java/org/figuramc/figura/lua/api/vanilla_model/VanillaModelAPI.java:
```diff
@@ -187,82 +187,88 @@ public class VanillaModelAPI {
     public final VanillaGroupPart PARROTS;
 
+    // Adds a VanillaPart to the list of all parts. Used for indexing
+    private <TPart extends VanillaPart>TPart addPart(TPart part) {
```
C# moment (usually in java (and more importantly, consistently in this codebase, *I think*) a template type is just a single letter, starting with `T`)

lenrik1589 commented on common/src/main/java/org/figuramc/figura/mixin/render/layers/CapeLayerMixin.java:
```diff
@@ -70,17 +70,17 @@ private void preRender(PoseStack poseStack, MultiBufferSource multiBufferSource,
+        // I'm pretty sure. This is due to it using the matrix stack instead of setting x,y,z,xRot,yRot,zRot on the parts.
+        // The cape functions completely differently than all other model parts of the player. Quite frankly,
+        // I don't want to deal with it any more than I already have, and I'm just going to leave this alone now and call it
+        // close enough.
```
changes, verry yes

lenrik1589 commented on common/src/main/java/org/figuramc/figura/mixin/render/layers/HumanoidArmorLayerMixin.java:
```diff
@@ -1,46 +1,75 @@
 import org.figuramc.figura.utils.RenderUtils;
+import org.spongepowered.asm.mixin.Final;
 import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Shadow;
```
i will trust imports here that they are not another Stray sequel

omoflop committed:
Merge branch 'FiguraMC:1.20' into 1.20

omoflop committed:
changed colors

lenrik1589 commented on common/src/main/java/org/figuramc/figura/wizards/WizardEntry.java:
```diff
@@ -0,0 +1,87 @@
+    public static WizardEntry register(String name, WizardEntry... dependencies) {
+        return register(name, Type.TOGGLE, dependencies);
+    }
+}
```
new line at the end of file

lenrik1589 commented on common/src/main/java/org/figuramc/figura/gui/widgets/BackendMessageWidget.java:
A little confused by this class, aren't there toasts?

lenrik1589 reviewed (commented):
a review, i guess

omoflop committed:
Cleaned up code

+ Removed stray imports
+ Motd uses TextUtils.tryParse instead of Component.Serializer.fromJson
+ Renamed BackendMessageWidget to BackendMotdWidget for clarity
+ Fixed a compiler error I accidentally pushed... oops
+ Removed white space after some comments

omoflop commented:
@lenrik1589 Fixed all that stuff, thanks for the feedback

omoflop committed:
fixed texture

+ texture made by april
